### ISSUE TITLE
PrototypeMemory and UPGDMemory core float32 overflow hardening and int32 bounds on dimensions

### ISSUE THREAD

**Issue by deepanshu-yd** [SlopDotCash/asi#441]
## Problem

`PrototypeMemoryConfig` in `alberta_framework/core/prototype_memory.py` and `UPGDMemoryConfig` in `alberta_framework/core/upgd_memory.py` lack float32 overflow validation on scalar hyperparameters (`update_rate`, `novelty_threshold`, `bandwidth`, `upgd_step_size`, multipliers, decays, thresholds, scales, adaptation rates, logit step sizes), allow adversarial float subclasses with out-of-domain exact ratios, and lack signed int32 bounds on `feature_dim`, `n_classes`, `slots_per_class`, `n_heads`, `hidden_sizes`, and warmup step counts.

## Reproduction

```python
from alberta_framework.core.prototype_memory import PrototypeMemoryConfig

# Accepts float32 overflow on update_rate
cfg = PrototypeMemoryConfig(feature_dim=4, n_classes=2, update_rate=1e100)
```

## Proposed Fix

- Use `round_real_to_float32_with_ratio` in `prototype_memory.py` and `upgd_memory.py` checking exact integer ratios, domain bounds, and rejecting class-property spoofing.
- Add `__post_init__` to dataclasses ensuring immediate validation on construction.
- Bound integer dimensions, counts, and sizes to signed int32 bounds (`2**31 - 1`).
- Add comprehensive test coverage in `tests/test_prototype_memory.py` and `tests/test_upgd_memory.py`.
